ecosystem/sep-direct-payment.md

```diff
@@ -0,0 +1,318 @@
+
+Each `fields` sub-object contains a key for each field name and an object with the following fields as the value:
+
+* `description`: description of field to show to user.
+* `choices`: list of possible values for the field.
+
+### Send
+
+#### Request
+
+```
+POST DIRECT_PAYMENT_SERVER/send
+Content-Type: application/json
+
+{
+  "amount": 100,
+  "require_receiver_info": ["address", "tax_id"],
+  "fields": {
+    "sender": {
+      "first_name": "Alice",
+      "last_name": "Jones"
+    },
+    "receiver": {
+      "first_name": "Bob",
+      "last_name": "Dillon",
+      "email_address": "bdillon@something.com"
+    },
+    "transaction": {
+      "routing_number": "442928834",
+      "account_number": "0029483242",
+      "type": "SEPA"
+    }
+  }
+}
+```
+
+This post requests attempts to initiate a payment through this anchor.  It should provide the amount and all the required fields (specified in the [`/info`](#info) endpoint).  The sending anchor can also provide a `require_receiver_info` object if it needs information on the receiver client that the receiving anchor can provide.
+
+If the request describes a valid transaction that this anchor can fulfill, we return a success response with details on what to send. If the request is not valid, or we need more info, we can return with an error response and expect the sending anchor to try again with updated values.
+
+##### Request Parameters
+
+Name | Type | Description
+-----|-----|------
+`require_receiver_info` | array | A list of [SEP-9](sep-0009.md) values requested for the receiving client
+`amount` | number | Amount of payment in destination currency
+`fields` | object | A key-pair object containing the values requested by the receiving anchor in their `/info` endpoint
+`lang` | string | (optional) Defaults to `en`. Language code specified using [ISO 639-1](https://en.wikipedia.org/wiki/ISO_639-1).  Any human readable error codes or field descriptions will be returned in this language.
+
+#### Responses
+
+##### Success (200 ok)
+This is the successful case where a receiving anchor confirms that they can fulfill this payment as described. The response body should be a JSON object with the following values
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Persistent identifier to check the status of this payment
+`stellar_account_id` | string | Stellar account to send payment to
+`stellar_memo_type` | string | Type of memo to attach to the stellar payment `(text | hash | id)`
+`stellar_memo` | string | The memo to attach to the stellar payment
+`receiver_info` | object | Key-value pairs of the information the sender requested via `require_receiver_info`
+
```

Comment:
I am curios here if there is a way for the receiving anchor to generate the stellar transaction id which should be used by the sending anchor when publish the transaction with the payment on the Stellar Network?
I am asking this because we plan to tie our AML checks on a Stellar Payment to a Stellar transaction id.

Reply:
There's no way to pre-generate a stellar tx id, but what you're describing is the reason for the stellar_memo field.  You can tie your AML checks or whatever you want to the stellar transaction that has the pre-generated memo.